vid_sdl.c

```diff
@@ -442,6 +442,26 @@ void VID_SetMouse(qbool fullscreengrab, qbool relative, qbool hidecursor)
 	{
 		vid_usinghidecursor = hidecursor;
 		SDL_ShowCursor( hidecursor ? SDL_DISABLE : SDL_ENABLE);
+		
+		//KleskBY //FIX: Sometimes cursor still visible in Windows and if you click game alt-tab. Is this code even safe?
+		Sleep(1);
+		int cursorState = SDL_ShowCursor(hidecursor ? SDL_DISABLE : SDL_ENABLE);
+		if (hidecursor)
+		{
+			while (cursorState != SDL_DISABLE)
+			{
+				cursorState = SDL_ShowCursor(SDL_DISABLE);
+				Sleep(1);
+			}
+		}
+		else
+		{
+			while (cursorState != SDL_ENABLE)
+			{
+				cursorState = SDL_ShowCursor(SDL_ENABLE);
+				Sleep(1);
+			}
+		}
```

Comment:
This code isn't safe because SDL_ShowCursor could return a negative number on failure and this isn't handled, and besides that, it locks the engine in a while loop as long as the cursor is or isn't hidden. This code is also redundant because you can do `SDL_ShowCursor(SDL_QUERY);` to get the state. Also, we use Sys_Sleep(), and the engine is only supposed to sleep at the end of each frame or at specific points in the server or client loop under certain conditions.

Reply:
Also `Sleep()` is windows-only, and I fixed various fullscreen window and mouse grab things cleanly.  If there's still an issue with cursor or window management please open a very specific issue and/or PR with details.